cub/cub/version.cuh

```diff
@@ -87,3 +87,5 @@
 static_assert(CUB_MAJOR_VERSION == CCCL_MAJOR_VERSION,"");
 static_assert(CUB_MINOR_VERSION == CCCL_MINOR_VERSION,"");
 static_assert(CUB_SUBMINOR_VERSION == CCCL_PATCH_VERSION,"");
+
+static_assert(CUB_VERSION == (CCCL_VERSION / 10), "Conflicting CUB and libcu++ versions detected. Check include paths.");
```

Comment:
important: This doesn't add any additional safety over the `static_assert`s above, does it? 

So maybe this means we already have the necessary safety checks in place, and the issues we've seen from mismatched versions originate from versions of these headers prior to these `static_assert`s being added.

Reply:
The one new check that this adds is in thrust

Reply:
I don't follow.

```
static_assert(THRUST_MAJOR_VERSION == CCCL_MAJOR_VERSION, "");
static_assert(THRUST_MINOR_VERSION == CCCL_MINOR_VERSION, "");
static_assert(THRUST_SUBMINOR_VERSION == CCCL_PATCH_VERSION, "");
```

is effectively redundant with

```
static_assert(THRUST_VERSION == (CCCL_VERSION / 10)
```

is it not? 

In other words, we'd never hit the new `static_assert` because one of the existing `static_assert`s would already fire.

Reply:
Yeah, I actually had another assert in that verified that cub and thrust have the same version, but pulled it out as that is transitive to the existing ones